Scenes/LocationMarker/LocationMarker.gd

```diff
@@ -14,3 +16,15 @@ func set_location_data(value : Resource) -> void:
 
 func _ready():
 	self.location_data = location_data
+	
+func _process(_delta):
+	if onCooldown and $CooldownTimer.time_left == 0:
+		onCooldown = false
+		$LocationMarkerBtn.disabled = false		
+		$AnimationPlayer.play("Active")
+
+func start_timer():
+	$LocationMarkerBtn.disabled = true
+	onCooldown = true
+	$AnimationPlayer.play("Cooldown")
+	$CooldownTimer.start()	
```

Comment:
You could possibly drop your use of `_process` and the `$CooldownTimer` by instead adding something like this to the end of `start_timer()`.
```
yield(get_tree().create_timer(10.0), "timeout")
$LocationMarkerBtn.disabled = false
$AnimationPlayer.play("Active")
```
`yield()` has some weird behavior, but it can be useful in cases like this.

Reply:
oh duh! I didn't even think about that!